src/GNDStk/defaulted.hpp

```diff
@@ -0,0 +1,72 @@
+
+// -----------------------------------------------------------------------------
+// defaulted
+// -----------------------------------------------------------------------------
+
+/*
+This class represents an "optional, with default" concept, for handling values
+in GNDS files that can appear optionally, but are considered to have particular
+defaults when they don't appear.
+
+T is, as you'd expect, the underlying type. A defaulted<T> contains a default
+value of type T, which *must* be provided in any constructor you call, as well
+as a std::optional<T> that optionally contains what we'll call a "hard value":
+a value that's considered to be other than the default.
+
+Note that you can give a defaulted<T> a hard value that happens to equal its
+default. If you do, then the value is considered to be a hard value - not the
+default. That is to say, the data structure does distinguish "has its default"
+versus "has a value that coincidentally happens to equal its default." It's an
+issue of how you construct and then modify (or don't modify) the object.
+*/
+
+template<class T>
+class defaulted {
```

Comment:
This should be called `Defaulted` as it is a class.

Reply:
I'm actually talking with Martin right now, and he was talking about making changes to the names to be more in line with our traditional style - at least with classes that are going to be publicly used. He wants to change that now, before he starts on the documentation next week.

Reply:
And that, for two big reasons: (1) so I don't have to modify the documentation later, and (2) so users don't start writing one thing, based on the documentation, and then have the rug pulled out from under them. :-)

Reply:
`defaulted` is definitely changing to `Defaulted`, and a few other things too.